src/app/api/fhir/proxy/[...api-path]/route.ts

```diff
@@ -1,18 +1,45 @@
-import { NextRequest } from "next/server";
+import { NextRequest, NextResponse } from "next/server";
 import { FhirProxy } from "@/integrations/fhir/FhirProxy";
 import { logRequest, logResponse } from "@/utils/loggerUtils";
+import { FhirSession } from "@/auth/fhir/FhirSession";
+import { FhirAuthError } from "@/auth/fhir/FhirAuthError";
+import { JwtVerificationInput } from "@/auth/fhir/JwtVerificationInput";
+
+const authErrorResponse = (text: string): Response => {
+    return new NextResponse(text, {status: 401});
+}
+
 
 const proxyHandler = async (request: NextRequest): Promise<Response> => {
-    logRequest(request)
+    const authHeader = request.headers.get("Authorization")
+    if(authHeader !== null) {
+        try {
+            // This verifies that incoming JWT is issued by a valid issuer and has a hpr-nummer claim
+            await FhirSession.fromVerifiedJWT(JwtVerificationInput.fromAuthorizationHeader(authHeader))
+        } catch (verifErr) {
+            if(verifErr instanceof FhirAuthError) {
+                return authErrorResponse(verifErr.message)
+            } else {
+                throw verifErr
+            }
+        }
+    } else {
+        authErrorResponse(`No Authorization header value in request. Cannot validate session`)
+    }
     const proxy = FhirProxy.initDipsProxy();
-    const resp = await proxy.forwardRequest(request)
+    return await proxy.forwardRequest(request)
+}
+
+const proxyHandlerWithRequestLogging = async (request: NextRequest): Promise<Response> => {
+    logRequest(request)
```

Comment:
kan det dukke opp sensitive data som ikke bør logges her?

Reply:
Godt spørsmål! 

Dette er jo ikkje noko som er endra i denne PR, men kan svare det eg tenkjer om det:

Viss du ser på logRequest funksjonen, så er det berre url som blir logga, og der er noko maskering av potensielt sensitive data, men den er ganske skjør, det er absolutt mulighet for at den ikke er god nok viss ulike Urler har sensitive data i seg. No trur eg ikkje nokon av fhir urlane vil ha så sensitive data i seg, det er vel berre interne id verdier som kan komme fram der.

Uansett trur eg ikkje det kjem så mykje an på, for eg trur at når vi skal rulle ut i prod vil trafikken der ikkje gå via denne proxyen uansett, men direkte mellom nettlesar og fhir server.

Eg meiner og at viss den skal gå via/frå vår server, så er det ikkje nødvendig å logge alle forespørsler og svar, så vi kan gjerne fjerne heile request logginga.

Reply:
Skjønner. Sensitiv data i URL vil uansett være en issue, og det må vi ta hvis vi kommer dit.
Bra nok for meg, takk for oppklarende svar!